database/migrations/2021_02_05_122930_migrate_to_utf8mb4.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Support\Facades\DB;
+
+class MigrateToUtf8mb4 extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        $this->migrateCharsetTo('utf8mb4', 'utf8mb4_unicode_ci');
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        $this->migrateCharsetTo('utf8', 'utf8_unicode_ci');
+    }
+
+    protected function migrateCharsetTo($charset, $collation)
+    {
+        $databaseName = DB::connection()->getDatabaseName();
+
+        // Change default charset and collation
+        DB::unprepared("ALTER SCHEMA {$databaseName} DEFAULT CHARACTER SET {$charset} DEFAULT COLLATE {$collation};");
+
+        // Get the list of all tables
+        $tableNames = DB::table('information_schema.tables')
+            ->where('table_schema', $databaseName)
+            ->get(['TABLE_NAME'])
+            ->pluck('TABLE_NAME');
+
+        // Iterate through the list and alter each table
+        foreach ($tableNames as $tableName) {
+            DB::unprepared("ALTER TABLE {$tableName} CONVERT TO CHARACTER SET {$charset} COLLATE {$collation};");
```

Comment:
This triggers fields to "upgrade", text > mediumtext etc, but it will be a catch-22 with our migrations, since if we change them to mediumtext then we get longtext...

>to avoid data type changes of the type just described, do not use CONVERT TO CHARACTER SET. Instead, use MODIFY to change individual columns. For example:

>ALTER TABLE t MODIFY latin1_text_col TEXT CHARACTER SET utf8;